File: round1068/pD.cpp
#include <bits/stdc++.h>

using namespace std;


vector<int> digs;
int aux(int i, int left, int prev, int score, int carry) {
    if (i == digs.size()) return score + left;
    if (carry == 1) {
        if (digs[i] == 1) return aux(i + 1, left, 0, score + 1, 1);
        int ans = aux(i + 1, left, 1, score, 0);
        if (left > 0) {
            ans = max(ans, aux(i + 1, left - 1, 0, score + 1, 1));
        }
        return ans;
    }
    int ans = aux(i + 1, left, digs[i], score, 0);
    if (prev == 0 && digs[i] == 1 && left > 0) {
        ans = max(ans, aux(i + 1, left - 1, 0, score + 1, 1));
    }
    return ans;
}

int main() {
    int t;
    cin >> t;
    for (int test = 0; test < t; test++) {
        int n, k;
        cin >> n >> k;
        digs.clear();
        for (;n > 0; n /= 2) {
            digs.push_back(n % 2);
        }

        cout << aux(0,k,0,0,0) << endl;
    }
}

File: round1068/pD.py
t = int(input())
for _ in range(t):
    n, k = [int(c) for c in input().split()]
    digs = [0] * 31
    for i in range(31):
        digs[i] = n % 2
        n //= 2
    # print("asdf", digs)
    # prev = 0
    # num = 0
    # for i in range(31):
    #     if prev == 0 and digs[i] == 1:
    #         num += 1
    #     prev = digs[i]

    # if num <= k:
    #     print(sum(digs) + k - 1)
    #     continue

    # dp = [[0] * 32 for _ in range(k + 1)]
    # for i in range(1, k + 1):
    #     for j in range(0, 31):


    
    def aux(i, left, prev=0, score=0, carry=0):
        if i == 31:
            return score + left
        global digs
        if carry == 1:
            if digs[i] == 1:
                return aux(i + 1, left, 0, score + 1, 1)
            else:
                ans = 0
                if left > 0:
                    ans = aux(i + 1, left - 1, 0, score + 1, 1)
                ans = max(ans, aux(i + 1, left, 1, score, 0))
                return ans
        ans = 0
        if prev == 0 and digs[i] == 1 and left > 0:
            ans = aux(i + 1, left - 1, 0, score + 1, 1)
        ans = max(ans, aux(i + 1, left, digs[i], score, 0))

        return ans
    print(aux(0, k))